src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -501,15 +501,22 @@ private void setupTestTemplateAndPush(Repository repository, Resource[] resource
             // First get files that are not dependent on the project type
             String templatePath = getProgrammingLanguageTemplatePath(programmingExercise.getProgrammingLanguage()) + "/test";
 
-            String projectTemplatePath = templatePath + "/projectTemplate/**/*.*";
+            // Java both supports Gradle and Maven as a test template
+            String projectTemplatePath = templatePath;
+            if (ProjectType.PLAIN_GRADLE.equals(programmingExercise.getProjectType()) || ProjectType.GRADLE_GRADLE.equals(programmingExercise.getProjectType())) {
```

Comment:
It might make sense to extract this and the related check for Maven to helper methods (Or maybe methods in the enum?), as these conditions are checked multiple times in this service and the BambooBuildPlanService.
Not strictly necessary, but it would increase the readability of the code IMO.

Reply:
I added methods to the enum.